File: app/src/main/java/com/zeyad/rxredux/screens/user/list/events/SearchUsersEvent.java
package com.zeyad.rxredux.screens.user.list.events;

import com.zeyad.rxredux.core.redux.BaseEvent;

/** @author by ZIaDo on 4/20/17. */
public class SearchUsersEvent implements BaseEvent {

    private final String query;

    public SearchUsersEvent(String s) {
        query = s;
    }

    public String getQuery() {
        return query;
    }
}

File: app/src/main/java/com/zeyad/rxredux/screens/user/list/events/SearchUsersEvent.kt
package com.zeyad.rxredux.screens.user.list.events

import com.zeyad.rxredux.core.BaseEvent
import kotlinx.android.parcel.Parcelize

@Parcelize
class SearchUsersEvent(private val query: String) : BaseEvent<String> {
    override fun getPayLoad(): String = query
}
